cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -4617,7 +4617,10 @@ const FunctionDoc binary_join_element_wise_doc(
      "emit a null (the default), are skipped, or replaced with a given string.\n"),
     {"*strings"}, "JoinOptions");
 
-const auto kDefaultJoinOptions = JoinOptions::Defaults();
+static const JoinOptions* GetDefaultJoinOptions() {
```

Comment:
These should be in the anonymous namespace already, so `static` is probably superfluous.

(same for other occurrences below)

Reply:
Done.